src/Domain/DomainEvent/UserCheckedIn.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Building\Domain\DomainEvent;
+
+use Prooph\EventSourcing\AggregateChanged;
+use Rhumsaa\Uuid\Uuid;
+
+final class UserCheckedIn extends AggregateChanged
+{
+    public static function toBuilding(Uuid $buildingId, string $username) : self
```

Comment:
Using a named constructor here prevents us from forgetting to pass parameters to the pre-existing `occur` named constructor.

Remember that events are immutable, so they cannot be changed after persistence if we messed them up, or if we forgot to store a required attribute.